### PR TITLE
GH-398 documentation, restructuring

### PR DESCRIPTION
**Pull request by danielmkn**
Project documentation has been restructured, similar to a new resource structure. 
Fixed links to the official documentation, minor errors, HCL formatting, trailing dots, etc. 
Addresses #422 and #398

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 6.5.3 (Apr 27, 2022). Tested on Artifactory 7.37.15
+
+IMPROVEMENTS:
+
+* reorganizing documentation, adding missing documentation links, fixing formatting. Nu changes in the functionality.
```
Comment (alexhung) [inline CHANGELOG.md:5]: Typo `Nu`

**Review by alexhung — CHANGES_REQUESTED** [review]
I know a lot of work has been put into this but the change in directory structure no longer follows Terraform's documentation: https://www.terraform.io/registry/providers/docs#directory-structure

I recommend we revert the directory structure.

**Review by chb0github — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (danielmkn) [reply CHANGELOG.md:5]: fixed, thanks!

**Comment by alexhung** [comment]
@danielmkn Before reverting the directory structure, you should reach out to Terraform and asks if the directory structure in their doc is set and not recommendation. Meaning if we use a different structure, would the documentation on TF registry page be broken?

File: docs/resource/repository/local/artifactory_local_debian_repository.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Artifactory Local Debian Repository Resource
```
Comment (alexhung) [inline docs/resource/repository/local/artifactory_local_debian_repository.md:1]: This file has not been moved to the new directory structure

**Review by maheshjfrog — DISMISSED** [review]

File: pkg/artifactory/resource/repository/remote/resource_artifactory_remote_java_repository.go [hunk 3]
```diff
@@ -13,13 +13,13 @@ func ResourceArtifactoryRemoteJavaRepository(repoType string, suppressPom bool)
 			Type:        schema.TypeBool,
 			Optional:    true,
 			Default:     false,
-			Description: `(Optional) When set, if a POM is requested, Artifactory attempts to fetch the corresponding jar in the background. This will accelerate first access time to the jar when it is subsequently requested. Default value is 'false'.`,
+			Description: `When set, if a POM is requested, Artifactory attempts to fetch the corresponding jar in the background. This will accelerate first access time to the jar when it is subsequently requested. Default value is 'false'.`,
```
Comment (alexhung) [inline pkg/artifactory/resource/repository/remote/resource_artifactory_remote_java_repository.go:16]: Why is `(Optional)` removed in this case? In other files, `(Optional)` is added to `Description`

File: docs/resource/repository/remote/remote.md [hunk 4]
```diff
@@ -14,49 +27,50 @@ resource "artifactory_remote_generic_repository" "my-remote-generic" {
...
   * `properties_enabled` - (Optional) If set, properties for artifacts that have been cached in this repository will be updated if they are modified in the artifact hosted at the remote Artifactory instance. The trigger to synchronize the properties is download of the artifact from the remote repository cache of the local Artifactory instance. Default value is 'false'.
-  * `source_origin_absence_detection` - (Optional) If set, Artifactory displays an indication on cached items if they have been deleted from the corresponding repository in the remote Artifactory instance. Default value is 'false'
-* `propagate_query_params` - (Optional, Default: false) When set, if query params are included in the request to Artifactory, they will be passed on to the remote repository.
+  * `source_origin_absence_detection` - (Optional) If set, Artifactory displays an indication on cached items if they have been deleted from the corresponding repository in the remote Artifactory instance. Default value is 'false'.
+* `propagate_query_params` - (Optional, Default: false) - When set, if query params are included in the request to Artifactory, they will be passed on to the remote repository.
```
Comment (alexhung) [inline docs/resource/repository/remote/remote.md:75]: Why is `-` added here?

File: docs/resource/repository/remote/artifactory_remote_pub_repository.md [hunk 5]
```diff
@@ -16,14 +15,16 @@ resource "artifactory_remote_pub_repository" "my-remote-pub" {
...
+Arguments have a one to one mapping with the [JFrog API](https://www.jfrog.com/confluence/display/RTF/Repository+Configuration+JSON).
+The following arguments are supported, along with the [common list of arguments for the remote repositories](remote.md):
 
-* `key` - (Required) The repository identifier. Must be unique system-wide
+* `key` - (Required) A mandatory identifier for the repository that must be unique. It cannot begin with a number or
+  contain spaces or special characters.
 * `description` - (Optional)
 * `notes` - (Optional)
-* `url` - (Required) - the remote repository URL.
+* `url` - (Required) the remote repository URL.
```
Comment (alexhung) [inline docs/resource/repository/remote/artifactory_remote_pub_repository.md:25]: Capitalization missing

File: docs/resource/repository/remote/artifactory_remote_go_repository.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+Arguments have a one to one mapping with the [JFrog API](https://www.jfrog.com/confluence/display/RTF/Repository+Configuration+JSON).
+The following arguments are supported, along with the [common list of arguments for the remote repositories](remote.md):
+
+* `key` - (Required) A mandatory identifier for the repository that must be unique. It cannot begin with a number or
+  contain spaces or special characters.
+* `description` - (Optional)
+* `notes` - (Optional)
+* `url` - (Required) The remote repo URL.
+* `vcs_git_provider` - Artifactory supports proxying the following Git providers out-of-the-box: GitHub or a remote Artifactory instance. Default value is "ARTIFACTORY".
```
Comment (alexhung) [inline docs/resource/repository/remote/artifactory_remote_go_repository.md:27]: Optional?

File: pkg/artifactory/resource/repository/remote/resource_artifactory_remote_java_repository.go (hunk 3 above)
Comment (danielmkn) [reply pkg/artifactory/resource/repository/remote/resource_artifactory_remote_java_repository.go:16]: Then I need to remove (Optional) from all .go files

File: docs/resource/repository/remote/remote.md (hunk 4 above)
Comment (danielmkn) [reply docs/resource/repository/remote/remote.md:75]: SHould be removed, missed.

Comment (danielmkn) [reply docs/resource/repository/remote/remote.md:75]: fixed

File: docs/resource/repository/remote/artifactory_remote_pub_repository.md (hunk 5 above)
Comment (danielmkn) [reply docs/resource/repository/remote/artifactory_remote_pub_repository.md:25]: thanks, fixed

File: docs/resource/repository/remote/artifactory_remote_go_repository.md (hunk 6 above)
Comment (danielmkn) [reply docs/resource/repository/remote/artifactory_remote_go_repository.md:27]: yes

**Review by alexhung — APPROVED** [review]
Too large to look at each file. I scan through them and looks ok.

**Pull request merged**